src/openforms/registrations/tasks.py

```diff
@@ -16,6 +17,56 @@
 logger = logging.getLogger(__name__)
 
 
+@app.task
+def pre_registration(submission_id: int) -> None:
+    submission = Submission.objects.get(id=submission_id)
+    if submission.pre_registration_completed:
+        return
+
+    registration_plugin = submission.get_registration_plugin()
+    if not registration_plugin:
+        submission.set_submission_reference()
+        submission.pre_registration_completed = True
+        submission.save()
+        return
+
+    options_serializer = registration_plugin.configuration_options(
+        data=submission.form.registration_backend_options
+    )
+
+    if not options_serializer.is_valid():
+        submission.set_submission_reference()
+        return
+
+    # If we are retrying, then an internal registration reference has been set. We keep track of it.
+    if submission.public_registration_reference:
+        if not submission.registration_result:
+            submission.registration_result = {}
+
+        assign(
+            submission.registration_result,
+            "temporary_internal_reference",
+            submission.public_registration_reference,
+            missing=dict,
+        )
+        submission.save()
```

Comment:
I'm also not sure if this should or should not be in an atomic transaction block :thinking: